lib/instruments.js

```diff
@@ -149,8 +150,8 @@ class Instruments {
     this.proc.stderr.setEncoding('utf8');
     let actOnStderr = (output) => {
       if (this.launchTimeout.afterSimLaunch && output && output.match(/CLTilesManagerClient: initialize/)) {
-        this.addSocketConnectTimer(this.launchTimeout.afterSimLaunch, 'afterLaunch', async () => {
-          await killAllInstruments();
+        this.addSocketConnectTimer(this.launchTimeout.afterSimLaunch, 'afterLaunch', async (done) => {
```

Comment:
After kill instruments process, it will trigger done() function 
In the case addSocketConnectTimer, it will call 
`this.launchResultDeferred.reject(new Error(ERR_NEVER_CHECKED_IN));`

```
addSocketConnectTimer (delay, type, doAction) {
    let socketConnectDelay = cancellableDelay(delay);
    socketConnectDelay.then(() => {
      return doAction(() => {
        log.warn(`Instruments socket client never checked in; timing out (${type})`);
        this.launchResultDeferred.reject(new Error(ERR_NEVER_CHECKED_IN));
      });
    }).catch(B.CancellationError, () => {}).done();
    this.socketConnectDelays.push(socketConnectDelay);
  }
```